modules/@apostrophecms/ui/ui/apos/mixins/AposCellMixin.js

```diff
@@ -0,0 +1,10 @@
+export default {
+  methods: {
+    // Fetch a field from the published version of the item, if available and
+    // defined, otherwise from the draft version of the item
+    displayValue(fieldName) {
```

Comment:
I don't think this logic should be in the cell. If the published doc has an empty field that has a value in the draft, we could end up showing the published title with draft values in other columns. We should be feeding all cells the same doc and doing this work on the manager level.

Reply:
We can't do that because some cells make different choices intentionally. Examples being the submitted properties (which are only available on the draft) and the cell context menu (which looks at lots of things). So I put the smarts in one place to be shared by the cells that made typical choices, but I didn't dare oversimplify for them.

Reply:
Also the submitted at / submitted by cells (those have a different problem right now, unrelated to this ticket but easier to fix when it's done).

Reply:
Those are more relevant because the context menu cell is separately called.

Reply:
How does this solve the issue of values showing from different doc versions?